### PR TITLE
ignore any leading/trailing whitespace in the peerlist

### PR DESCRIPTION
**Pull request by Geometer1729**
Cherry-pick #15739 into compatible

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them already closed

### CODE REVIEW

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Review by svv232 — APPROVED** [review]

File: src/lib/mina_net2/multiaddr.ml [hunk 1]
```diff
@@ -39,6 +39,7 @@ let valid_as_peer t =
 
 let of_file_contents contents : t list =
   String.split ~on:'\n' contents
+  |> List.map ~f:String.strip
```
Comment (mrmr1993) [inline src/lib/mina_net2/multiaddr.ml:42]: Nit: This would be more efficient as a combined `List.filter_map`

**Review by mrmr1993 — APPROVED** [review]

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Comment by Geometer1729** [comment]
!ci-build-me

**Pull request merged**